backend/app/routers/experiment_templates.py

```diff
@@ -241,40 +256,44 @@ def find_specific_experiment_templates(
                 headers={"WWW-Authenticate": "Bearer"},
             )
 
-    if only_finalized:
-        search_conditions.append(ExperimentTemplate.state == TemplateState.FINISHED)
-    if only_not_archived:
-        search_conditions.append(ExperimentTemplate.is_archived == False)  # noqa: E712
-    if only_public:
-        search_conditions.append(ExperimentTemplate.is_public == True)  # noqa: E712
-
-    if len(search_conditions) > 0:
-        multi_query = (
-            operators.Or(*search_conditions)
-            if query_operator == QueryOperator.OR
-            else operators.And(*search_conditions)
+    if finalized is not None:
+        search_conditions.append(
+            get_compare_operator_fn(finalized)(
+                ExperimentTemplate.state, TemplateState.FINISHED
+            )
         )
-        return ExperimentTemplate.find(multi_query, **page_kwargs)
-
-    return ExperimentTemplate.find_all(**page_kwargs)
+    if approved is not None:
+        search_conditions.append(ExperimentTemplate.approved == approved)
+    if archived is not None:
+        search_conditions.append(ExperimentTemplate.archived == archived)
+    if public is not None:
+        search_conditions.append(ExperimentTemplate.public == public)
+
+    multi_query = (
+        operators.Or(*search_conditions)
```

Comment:
The initial condition regarding accessibility should be always required. We can not concatenate it with `OR` like this, I assume.